server/broadcast.go

```diff
@@ -542,9 +544,16 @@ func (bsm *BroadcastSessionsManager) chooseResults(submitResultsCh chan *SubmitR
 			if untrustedResult.Err == nil {
 				bsm.sessionVerified(untrustedResult.Session)
 			}
+			// suspend sessions which returned incorrect results
+			for _, s := range sessionsToSuspend {
+				bsm.suspendAndRemoveOrch(s)
+			}
 			return untrustedResult.Session, untrustedResult.TranscodeResult, untrustedResult.Err
-		} else if monitor.Enabled {
-			monitor.FastVerificationFailed()
+		} else {
```

Comment:
Not related to this exact issue or very high priority, so maybe we should open a new issue/PR for this:

Maybe we should reset our `verifiedSession` to `nil` if the hash it sends doesn't match?

```suggestion
		} else {
			if (bsm.verifiedSession == untrustedResult.Session) {
				bsm.verifiedSession = nil
			}
```

Reply:
I see one issue with adding this condition. If we add it, then a false-negative will cause `bsm.verifiedSession = nil` and in a result, we'll start looking for a new session. That can in turn cause bouncing between sessions if we encounter many false-negative scenarios.

We decided that we'll suspend sessions only if it failed p-hash verification but at the same time the other untrusted sessions passed the verification. We could do the same and maybe set `verifiedSession` to `nil` when we suspend the session, but technically it does not matter, because [this condition](https://github.com/livepeer/go-livepeer/blob/e2f94ca538a5bfe6b9ac9fe49848fb4b9536e135/server/broadcast.go#L434) takes into consideration both `bsm.verifiedSession = nil` and excluding suspended sessions.

Reply:
Good point about false negatives causing swaps, and yeah we'd anyway suspend a verified session and look for a new one if the other untrusted session passes. LGTM feel free to merge :+1: